Libraries/Opc.Ua.PubSub/Uadp/UdpClientCreator.cs

```diff
@@ -0,0 +1,296 @@
+/* ========================================================================
+ * Copyright (c) 2005-2020 The OPC Foundation, Inc. All rights reserved.
+ *
+ * OPC Foundation MIT License 1.00
+ * 
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * The complete license agreement can be found here:
+ * http://opcfoundation.org/License/MIT/1.00/
+ * ======================================================================*/
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+namespace Opc.Ua.PubSub.Uadp
+{
+    /// <summary>
+    /// Where is a method call used in 
+    /// </summary>
+    public enum UsedInContext
+    {
+        /// <summary>
+        /// Publisher context call
+        /// </summary>
+        Publisher,
+        /// <summary>
+        /// Subscriber context call
+        /// </summary>
+        Subscriber
+    };
+
+    /// <summary>
+    /// Specialized in creating the necessary <see cref="UdpClient"/> instances from an URL
+    /// </summary>
+    internal class UdpClientCreator
+    {
+
+        public const int SIO_UDP_CONNRESET = -1744830452;
+
+        /// <summary>
+        /// Parse the url into an IPaddress and port number
+        /// </summary>
+        /// <param name="url"></param>
+        /// <returns>A new instance of <see cref="IPEndPoint"/> or null if invalid URL.</returns>
+        internal static IPEndPoint GetEndPoint(string url)
+        {
+            Uri connectionUri;
+            if (url != null && Uri.TryCreate(url, UriKind.Absolute, out connectionUri))
+            {
+                if (connectionUri.Scheme != "opc.udp")
```

Comment:
maybe define this udp aside UriSchemeOpcTcp in Core.Utils.

Reply:
done. defined public _const string UriSchemeOpcUdp = "opc.udp";_ in Utils.cs